java/src/jmri/jmrix/dccpp/serial/DCCppAdapter.java

```diff
@@ -151,6 +156,28 @@ public DataOutputStream getOutputStream() {
         return null;
     }
 
+    /**
+     * Set up the keepAliveTimer, and start it.
+     */
+    private void keepAliveTimer() {
+        if (keepAliveTimer == null) {
+            keepAliveTimer = new java.util.TimerTask(){
+                    @Override
+                    public void run() {
+                        // If the timer times out, send a request for status
+                        DCCppAdapter.this.getSystemConnectionMemo().getDCCppTrafficController()
+                            .sendDCCppMessage(
+                                              jmri.jmrix.dccpp.DCCppMessage.makeCSStatusMsg(),
+                                              null);
+                    }
+                };
+        } else {
+            keepAliveTimer.cancel();
+        }
+        jmri.util.TimerUtil.schedule(keepAliveTimer, keepAliveTimeoutValue, keepAliveTimeoutValue);
```

Comment:
I'm afraid that this will not work.

https://docs.oracle.com/javase/7/docs/api/java/util/Timer.html#schedule(java.util.TimerTask,%20java.util.Date)
Quote from the Timer javadoc:
`IllegalStateException - if task was already scheduled or cancelled, timer was cancelled, or timer thread terminated.`

So if the timer task has been cancelled, can it then be scheduled again? I'm not sure about that.
I think you always need to create a new timer task.

Reply:
Thanks for the feedback, but I copied that verbatim from DCCppEthernetAdapter. I've tested it and it works fine.

Reply:
Thanks.